lib/models/Player.js

```diff
@@ -598,10 +598,40 @@ Player.prototype.browse = function browse(objectId, startIndex, limit) {
     });
 };
 
+Player.prototype.browseAll = function browseAll(objectId) {
+  let result = {
+    items: [],
+    startIndex: 0,
+    numberReturned: 0,
+    totalMatches: 1
+  };
+
+  let getChunk = (chunk) => {
+    Array.prototype.push.apply(result.items, chunk.items);
+    result.numberReturned += chunk.numberReturned;
+    result.totalMatches = chunk.totalMatches;
+
+    if (chunk.startIndex + chunk.numberReturned >= chunk.totalMatches) {
+      return result;
+    }
+
+    // Recursive promise chain
+    return this.browse(objectId, chunk.startIndex + chunk.numberReturned, 0)
+        .then(getChunk);
+  };
+
+  return Promise.resolve(result)
+      .then(getChunk);
+};
+
```

Comment:
You could consider returning only items here, since the result objects primary use was pagination. Since you would return the complete result here, it is of small use